privacy-protections/storage-partitioning/helpers/commonTests.js

```diff
@@ -0,0 +1,552 @@
+/* exported testAPIs */
+/* globals FIRST_PARTY_HOSTNAME */
+
+const timeout = 1000; // ms; used for cross-tab communication APIs
+
+function getURL (path, fileType, key) {
+    const url = new URL(`/partitioning/${path}`, window.location.origin);
+    url.searchParams.set('fileType', fileType);
+    url.searchParams.set('key', key);
+    return url;
+}
+
+const sleepMs = (timeMs) => new Promise(
+    (resolve, reject) => setTimeout(resolve, timeMs)
+);
+
+const loadSubresource = async (tagName, url) => {
+    const element = document.createElement(tagName);
+    document.body.appendChild(element);
+    const resultPromise = new Promise((resolve, reject) => {
+        element.addEventListener('load', resolve, { once: true });
+        element.addEventListener('error', reject, { once: true });
+    });
+    element.src = url;
+    try {
+        return await resultPromise;
+    } catch (e) {
+        // some sort of loading error happened
+        return e;
+    }
+};
+
+// Validates storage APIs which return the value of the token stored in
+// the storage container within each test context. E.g.,
+//
+// document.cookie
+//     same-site: [ { "value": "51c69e1b" }, { "value": "51c69e1b" } ]
+//    cross-site: [ { "value": "fd10f7f5d2cf" }, { "value": "fd10f7f5d2cf" } ]
+function validateStorageAPI (sameSites, crossSites, reference, random) {
+    if (
+        (sameSites.every(v => v.error === 'Unsupported')) &&
+        (crossSites.every(v => v.error === 'Unsupported'))
+    ) {
+        return 'unsupported';
+    } else if (
+        (sameSites.every(v => v.error && v.error.endsWith('is deprecated'))) &&
+        (crossSites.every(v => v.error && v.error.endsWith('is deprecated')))
```

Comment:
not sure if I'm reading it correctly, but if API is deprecated (e.g. websql) it still can be used and we don't know when this will go away, so I would ignore the fact that it's deprecated in testing

Reply:
Good catch! I'll take another look at this. I had assumed that a value wasn't returned, but I'll double check. See: https://github.com/duckduckgo/privacy-test-pages/blob/4fd601c3ac15cc6d50812b0a7ef68e16f3d39ef6/privacy-protections/storage-partitioning/helpers/commonTests.js#L153-L158

Reply:
Looks like this is no longer necessary to special case.